Dockerfile

```diff
@@ -130,21 +130,21 @@ ENV MIPS_ELF_ROOT /opt/mips-mti-elf/2016.05-03
 ENV PATH ${PATH}:${MIPS_ELF_ROOT}/bin
 
 # Install RISC-V binary toolchain
+ARG RISCV_VERSION=8.2.0-2.2-20190521
+ARG RISCV_BUILD=0004
 RUN mkdir -p /opt && \
-        wget -q https://github.com/gnu-mcu-eclipse/riscv-none-gcc/releases/download/v7.2.0-2-20180110/gnu-mcu-eclipse-riscv-none-gcc-7.2.0-2-20180111-2230-centos64.tgz -O- \
+        wget -q https://github.com/gnu-mcu-eclipse/riscv-none-gcc/releases/download/v${RISCV_VERSION}/gnu-mcu-eclipse-riscv-none-gcc-${RISCV_VERSION}-${RISCV_BUILD}-centos64.tgz -O- \
         | tar -C /opt -xz && \
     echo 'Removing documentation' >&2 && \
-    rm -rf /opt/gnu-mcu-eclipse/riscv-none-gcc/*/share/doc && \
+      rm -rf /opt/gnu-mcu-eclipse/riscv-none-gcc/*/share/doc && \
     echo 'Deduplicating binaries' >&2 && \
     cd /opt/gnu-mcu-eclipse/riscv-none-gcc/*/riscv-none-embed/bin && \
-    for f in *; do rm "$f" && ln "../../bin/riscv-none-embed-$f" "$f"; done && cd -
+      for f in *; do test -f "../../bin/riscv-none-embed-$f" && \
+       ln -f "../../bin/riscv-none-embed-$f" "$f"; \
+      done && \
+    cd -
 
-# HACK download arch linux' flex dynamic library
-RUN wget -q https://sgp.mirror.pkgbuild.com/core/os/x86_64/flex-2.6.4-2-x86_64.pkg.tar.xz -O- \
```

Comment:
Why was this needed before and why is is not needed anymore?

Reply:
See #39, libfl on ubuntu is somehow weird. the riscv toolchain could not use it. thus we used arch linux' libfl package.
The current risc-v toolchain comes bundled with libfl, thus the hack is not needed anymore.